src/commands/infra/generate.ts

```diff
@@ -180,61 +195,63 @@ export const validateTemplateSources = async (
+      `Provided source in template directory was not found, attempting to clone the template source repo locally.`
+    );
+    createGenerated(sourcePath);
+  } else {
+    logger.info(
+      `Source template folder found. Validating existence of repository.`
+    );
+  }
+  // Checking for git remote
+  const result = await simpleGit(sourcePath).listRemote([source]);
+  if (!result) {
+    logger.error(result);
+    throw new Error(
+      `Unable to clone the source remote repository. \
+The remote repo may not exist or you do not have the rights to access it`
+    );
+  }
+
+  logger.info(`Remote source repo: ${safeLoggingUrl} exists.`);
+  logger.info(
+    `Checking if source repo: ${safeLoggingUrl} has been already cloned to: ${sourcePath}.`
+  );
+
   try {
-    if (!fs.existsSync(sourcePath)) {
-      logger.warn(
-        `Provided source in template directory was not found, attempting to clone the template source repo locally.`
-      );
-      await createGenerated(sourcePath);
+    // Check if .git folder exists in ${sourcePath}, if not, then clone
+    // if already cloned, 'git pull'
+    if (fs.existsSync(path.join(sourcePath, ".git"))) {
+      // Make sure we have the latest version of all releases cached locally
+      await simpleGit(sourcePath).fetch("all");
+      await simpleGit(sourcePath).pull("origin", "master");
+      logger.info(`${safeLoggingUrl} already cloned. Performing 'git pull'...`);
     } else {
-      logger.info(
-        `Source template folder found. Validating existence of repository.`
-      );
-    }
-    // Checking for git remote
-    const result = await simpleGit(sourcePath).listRemote([source]);
-    if (!result) {
-      logger.error(
-        `Unable to clone the source remote repository. Does the remote repo exist? Do you have the rights to access it? ${result}`
-      );
-      return false;
-    } else {
-      logger.info(`Remote source repo: ${safeLoggingUrl} exists.`);
-      logger.info(
-        `Checking if source repo: ${safeLoggingUrl} has been already cloned to: ${sourcePath}.`
-      );
-      // Check if .git folder exists in ${sourcePath}, if not, then clone
-      // if already cloned, 'git pull'
-      if (fs.existsSync(path.join(sourcePath, ".git"))) {
-        // Make sure we have the latest version of all releases cached locally
-        await simpleGit(sourcePath).fetch("all");
-        await simpleGit(sourcePath).pull("origin", "master");
-        logger.info(
-          `${safeLoggingUrl} already cloned. Performing 'git pull'...`
-        );
-      } else {
-        await gitClone(source, sourcePath);
-      }
-      // Checkout tagged version
-      logger.info(`Checking out template version: ${version}`);
-      await simpleGit(sourcePath).checkout(version);
+      await gitClone(source, sourcePath);
     }
+    // Checkout tagged version
+    logger.info(`Checking out template version: ${version}`);
+
+    await simpleGit(sourcePath).checkout(version);
```

Comment:
Something in this step is returning an empty message:
<img width="804" alt="Screen Shot 2020-01-28 at 9 28 17 AM" src="https://user-images.githubusercontent.com/16811248/73288844-93ec2e00-41b0-11ea-8136-1aae2c371f49.png">

This happens when I run in a hierarchy where there is no parent definition.yaml and a there exists a leaf definition.yaml.

Reply:
this empty info: liner will go away after this PR is merged https://github.com/CatalystCode/spk/pull/214.
The case where parent definition.yaml is missing not not handled. You, Nate and Andre (who are the domain experts) can to decide what to do. And it can be handle in a different PR (not making this PR too big).

Reply:
Seems like that PR is merged, and I am still seeing that empty line.

Reply:
this is going to change again. see PR: https://github.com/CatalystCode/spk/pull/221

Reply:
Thanks for the update @dennisseah

Reply:
I see that is it is resolved now by appending `--end log: 0--` at the end of the log. Tested.